sql/analyzer/stored_procedures.go

```diff
@@ -378,7 +378,12 @@ func applyProceduresCall(ctx *sql.Context, a *Analyzer, call *plan.Call, scope *
 
 	procedure := scope.procedures.Get(ctx.GetCurrentDatabase(), call.Name, len(call.Params))
 	if procedure == nil {
-		return nil, transform.SameTree, sql.ErrStoredProcedureDoesNotExist.New(call.Name)
+		err := sql.ErrStoredProcedureDoesNotExist.New(call.Name)
+		if ctx.GetCurrentDatabase() == "" {
+			cause := errors.NewKind("this might be because no database is selected")
+			err = sql.ErrStoredProcedureDoesNotExist.Wrap(cause.New(), call.Name)
```

Comment:
```suggestion
			err = fmt.Errorf("%w; this might be because no database is selected", err)

```

Reply:
Nice! thanks for the feedback on making this error code look more idiomatic. Updated.